crates/subspace-networking/src/create.rs

```diff
@@ -30,6 +38,153 @@ use tracing::info;
+            // For each source peer ID one circuit every 2 minute with up to 30 circuits per hour.
+            circuit_src_rate_limit_per_peer: rate_limiter::GenericRateLimiterConfig {
+                limit: NonZeroU32::new(30).expect("30 > 0"),
+                interval: Duration::from_secs(60 * 2),
+            },
+            // For each source IP address one circuit every minute with up to 60 circuits per hour.
+            circuit_src_rate_limit_per_ip: rate_limiter::GenericRateLimiterConfig {
+                limit: NonZeroU32::new(60).expect("60 > 0"),
+                interval: Duration::from_secs(60),
+            },
+        }
+    }
+}
+
+impl RelayLimitSettings {
+    /// Converts `RelayLimitSettings` to libp2p `relay::Config`.
+    pub fn to_relay_config(self) -> RelayConfig {
+        let reservation_rate_limiters = vec![
+            rate_limiter::new_per_peer(self.reservation_rate_limit_per_peer),
+            rate_limiter::new_per_ip(self.circuit_src_rate_limit_per_ip),
+        ];
+
+        let circuit_src_rate_limiters = vec![
+            rate_limiter::new_per_peer(self.circuit_src_rate_limit_per_peer),
+            rate_limiter::new_per_ip(self.circuit_src_rate_limit_per_ip),
+        ];
+
+        RelayConfig {
+            max_reservations: self.max_reservations,
+            max_reservations_per_peer: self.max_circuits_per_peer,
+            reservation_duration: self.reservation_duration,
+            reservation_rate_limiters,
+            max_circuits: self.max_circuits,
+            max_circuits_per_peer: self.max_circuits_per_peer,
+            max_circuit_duration: self.max_circuit_duration,
+            max_circuit_bytes: self.max_circuit_bytes,
+            circuit_src_rate_limiters,
+        }
+    }
+}
+
+/// Defines relay circuits configuration for the networking.
+#[derive(Clone, Debug)]
+pub enum RelayConfiguration {
+    /// Relay server configuration. The node will be able to listen on the provided Multiaddr with
+    /// adding it as an external address.
+    /// Expected format for Multiaddr: /ip4/127.0.0.1/tcp/50000
+    /// The relay configuration will be configured with provided RelayLimitSettings.
+    /// This option will toggle the relay behaviour (relay server).
+    Server(Multiaddr, RelayLimitSettings),
+
+    /// Relay-client configuration in accepting mode (server behind NAT). It will enable listening
+    /// on the provided Multiaddr with registering a circuit reservation.
+    /// Expected format for Multiaddr: /ip4/127.0.0.1/tcp/50000/p2p/<server_peer_id>/p2p-circuit
+    /// This option will toggle the relay-client behaviour.
+    ClientAcceptor(Multiaddr),
+
+    /// Relay-client configuration in requesting mode (a client for a server behind NAT).
+    /// It will enable creating a relay circuit.
+    /// Expected format for Multiaddr: /ip4/127.0.0.1/tcp/50000/p2p/<server_peer_id>/p2p-circuit
+    /// This option will toggle the relay-client behaviour.
+    ClientInitiator,
```

Comment:
I find these explanations confusing.

First of all, server behind NAT is strange, usually it is clients that are behind NAT, not servers.

Second is `/ip4/127.0.0.1/tcp/50000/p2p/<server_peer_id>/p2p-circuit`, usually `p2p-circuit` isn't the end, there is `/p2p/QmAlice` after it.

Third, there is expected multiaddr format description for `ClientInitiator`, but no such parameter in it (likely copy-paste).